docs/engine.md

```diff
@@ -50,6 +50,8 @@ condition reference that cannot be resolved an exception is thrown. Turning
 this option on will cause the engine to treat unresolvable condition references
 as failed conditions. (default: false)
 
+`allowFactsInEventParams` - By default when rules succeed or fail the events emitted are clones of the event in the rule declaration. When setting this option to true the parameters on the events will be have any fact references resolved. (default: false)
```

Comment:
maybe `replaceFactsInEventParams` is more accurate, since there's a mutation happening. Just a thought.

Reply:
We clone the event so it's technically not "mutating" what has been passed to the engine. However I do think replace is a better verb for this.